File: Day01/puzzle1.py
if 0:
    INPUT_FILE = "input_simple.txt"
else:
    INPUT_FILE = "input.txt"
    
# LAZY implementation
    
calories_list: list[list] = []


with open(INPUT_FILE, "r") as f:
    calories: list[int] = []
    for line in f.readlines():
        if line == "\n":
            calories_list.append(calories)
            calories = []
        else:
            calories.append(int(line))
    calories_list.append(calories)

calories_list_int: list[int] = []

for cal_list in calories_list:
    n = 0
    for cal in cal_list:
        n += cal
    calories_list_int.append(n)
    
print(max(calories_list_int))
    

File: Day01/puzzle1.c
#include "day01_helper.c"

#if 0
#define INPUT_FILE "input_simple.txt"

#else
#define INPUT_FILE "input.txt"

#endif

int main() {
    IntArray callories;
    int err = load_file(&callories, INPUT_FILE);
    if (err) return err;

    int max;
    int_array_max(&callories, &max);

    printf("%d", max);

    int_array_clear(&callories);
    return 0;
}